protocol/transaction.go

```diff
@@ -0,0 +1,40 @@
+package protocol
+
+// transaction represents a sequence of frontend and backend messages
+// that apply only on commit. the purpose of transaction is to support
+// extended query flow.
+type transaction struct {
+	p   *Protocol
+	in  []Message
+	out []Message
+}
+
+// Read uses Protocol to read the next message into the transaction's incoming messages buffer
+func (t *transaction) Read() (msg Message, err error) {
+	if msg, err = t.p.read(); err == nil {
+		t.in = append(t.in, msg)
```

Comment:
why do we need to keep the message? shouldn't we simply `return t.p.read()`?

Reply:
I think that operations we'll add later might need previous messages of the current transaction